Automattic/newspack-plugin
# Business setup for onboarding wizard

### All Submissions:






### Changes proposed in this Pull Request:



Closes #54 .

This PR implements an initial onboarding wizard for setting up general site settings. This first version handles WooCommerce address info and Stripe settings. If we like the toggle for the Stripe settings, we'll need to add a Muriel toggle and migrate the core toggle used here to a Muriel one in the future.

Screenshots in details
<details>

<img width="579" alt="Screen Shot 2019-06-03 at 11 36 59 AM" src="https://user-images.githubusercontent.com/7317227/58827456-eceecc80-85f7-11e9-9332-09ea61aa6f84.png">
<img width="564" alt="Screen Shot 2019-06-03 at 12 12 19 PM" src="https://user-images.githubusercontent.com/7317227/58827847-e3199900-85f8-11e9-8ef6-a12b2ee1da06.png">
<img width="569" alt="Screen Shot 2019-06-03 at 12 14 16 PM" src="https://user-images.githubusercontent.com/7317227/58827978-34298d00-85f9-11e9-8143-f7165de16a65.png">
<img width="558" alt="Screen Shot 2019-06-03 at 11 37 07 AM" src="https://user-images.githubusercontent.com/7317227/58827458-eceecc80-85f7-11e9-90c8-ff7b4eb898d1.png">

</details>

### Questions:
- Does it make sense to have these screens as part of the general onboarding wizard, or should these screens be a separate wizard since they only apply to sites with memberships (the first wizard in the Memberships checklist)?
- At what point should we tweak the WooCommerce settings for memberships sites? For example, disable coupons and product reviews. Currently that's done as part of adjusting the location settings, but that seems like a weird place to do that.
- @sonjaleix which Primary Button spacing do you prefer? Near the last field like the location settings screenshot, or with a little space between the button and the fields like the Stripe settings screenshot. I'll standardize it between the screens when I know which is best.

### How to test the changes in this Pull Request:

1. `npm run clean; npm run build`. **Make sure you have WooCommerce active**. We can adjust the logic once the wizard is fleshed out further to take into account sites that don't want to use WooCommerce, but the wizard wouldn't have anything to set up without WC so I'll do that in a future PR.
2. Navigate to Newspack > Onboarding in WP Admin, and go through the wizard.
3. This is the first wizard that will do a just-in-time install of a necessary plugin, so you can test that. Delete the Stripe plugin and toggle Stripe enabled on the wizard's Stripe settings. Stripe should be installed and activated when you click the 'Finish' button. If enabled is not toggled, Stripe should not be installed or activated.

### Other information:
